### PR TITLE
Fixes openshift-tests dry-run failure asking for KUBERNETES_MASTER

### PR DESCRIPTION
**Pull request by Rajalakshmi-Girish**
Fixes https://github.com/openshift/origin/issues/28159

### CODE REVIEW

File: pkg/cmd/openshift-tests/run/flags.go [hunk 1]
```diff
@@ -75,10 +76,14 @@ func (f *RunSuiteFlags) SetIOStreams(streams genericclioptions.IOStreams) {
 
 func (f *RunSuiteFlags) ToOptions(args []string) (*RunSuiteOptions, error) {
 	adminRESTConfig, err := kubeconfig.GetStaticRESTConfig()
```
Comment (vrutkovs) [inline pkg/cmd/openshift-tests/run/flags.go:78]:
```suggestion
	if f.GinkgoRunSuiteOptions.DryRun {
	    return nil
	}
	adminRESTConfig, err := kubeconfig.GetStaticRESTConfig()
```

File: pkg/cmd/openshift-tests/run/flags.go [hunk 2]
```diff
@@ -75,10 +76,14 @@ func (f *RunSuiteFlags) SetIOStreams(streams genericclioptions.IOStreams) {
...
-	if err != nil {
+	if err != nil && !f.GinkgoRunSuiteOptions.DryRun {
 		return nil, err
 	}
 
+	if err != nil && f.GinkgoRunSuiteOptions.DryRun {
+		adminRESTConfig = &rest.Config{}
+	}
+
```
Comment (vrutkovs) [inline pkg/cmd/openshift-tests/run/flags.go:86]:
```suggestion
	switch {
	case err != nil && f.GinkgoRunSuiteOptions.DryRun:
		fmt.Fprintf(f.ErrOut, "Unable to get admin rest config, skipping apigroup check in the dry-run mode: %v\n", err)
		adminRESTConfig = &rest.Config{}
	case err != nil && !f.GinkgoRunSuiteOptions.DryRun:
		return nil, fmt.Errorf("unable to admin rest config, skipping apigroup check in the dry-run mode: %w", err)
```

Comment (Rajalakshmi-Girish) [reply pkg/cmd/openshift-tests/run/flags.go:86]: @vrutkovs Isn't it a wrong message in line 84 that you suggested?
The condition in line 83 above says !DryRun, whereas the error message has the text `skipping apigroup check in the dry-run mode`
I suppose the Error message at https://github.com/openshift/origin/blob/14db7484bdd1a1e0d97f1d806905ccd04a237778/pkg/clioptions/suiteselection/suite_flags.go#L114-L115 is also wrong.

Comment (vrutkovs) [reply pkg/cmd/openshift-tests/run/flags.go:86]: Ah, right, sorry, I copypasted it. Feel free to adjust the error message when applying the suggestion and squash commits

**Comment by Rajalakshmi-Girish** [comment]
I see all the failing checks to be e2e runs on aws. They don't seem to be related to the change in this PR.
The error log in jobs mostly says `unable to create pod: etcdserver: mvcc: database space exceeded]`

Also, I don't think I have permission to retest the failed check.
@vrutkovs / @bparees Can you please look into this?

**Comment by vrutkovs** [comment]
```
 INFO[2023-08-16T04:24:56Z] Running `gofmt -s -l` on 649 file(s).
./pkg/cmd/openshift-tests/run/flags.go
verify-gofmt failed - please run `make update-gofmt` 
```
this certainly needs to be fixed, the rest are flakes

**Pull request merged**